linked_list/remove_dupes/main.c

```diff
@@ -0,0 +1,67 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include "linked_list.h"
+
+// This program defines a function to remove duplicate elements from a linked list
+
+void remove_dupes(struct node* head);
+
+int main()
+{
+    struct node* ll;
+    struct node* sll;
+    ll = linked_list(7);
+    int i;
+    for (i = 1; i < 10; i++){
+        push(ll, 7);
+    }
+    printf("First list...\n");
+    print_linked_list(ll);
+    printf("Removing duplicates ... \n\n");
+    remove_dupes(ll);
+    print_linked_list(ll);
+    sll = linked_list(0);
+    for (i=1; i<10; i++){
+        push(sll,(i%3));
+    }
+    printf("\n\nSecond linked list...\n");
+    print_linked_list(sll);
+    printf("Removing duplicates ... \n\n");
+    remove_dupes(sll);
+    print_linked_list(sll);
+    return 0;
+}
+
+/*
+ * remove_dupes - Removes duplicate elements from a linked list in O(n^2) time using O(1) space.
+ *                This will be revisited and improved once i learn how to implement hash tables in C.
+ * @head: pointer to first node of linked list
+ */
+void remove_dupes(struct node* head){
+    struct node* tmp;
+    struct node* tmp2;
+    struct node* tmp3;
+    tmp = head;
+    tmp2 = head;
+    while (tmp != NULL && tmp->next != NULL){
+        tmp2 = tmp;
+        while (tmp2->next != NULL){
```

Comment:
It might be better for this inner while loop to be replaced by a helper function to make the logic easier to understand.